compiler/damlc/tests/daml-test-files/OutOfMemory.daml

```diff
@@ -0,0 +1,12 @@
+-- Copyright (c) 2022, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @ENABLE-SCENARIOS
+
+module OutOfMemory where
+
+main = scenario do
+  loop ()
+
+loop : Action m => () -> m ()
```

Comment:
After adding this test, I noticed that the `OutOfMemory` error isn't caught. Looking into it with @remyhaemmerle-da, he realized that's because the JVM can't reclaim the memory from the `Speedy.Machine` since that's an argument to `ScenarioRunner`. Instead, we should construct the machine inside `ScenarioRunner`, but that's a bigger rewrite than I wanted for this PR

Reply:
If the memory is being used by the continuation stack, then you could call `setExpressionToEvaluate` (which calls `initialKontStack`) which should release it.